mos_tests/neutron/python_tests/test_dvr.py

```diff
@@ -253,3 +259,344 @@ def test_routing_after_reset_computes(self, prepare_openstack, env_name):
         self.check_ping_from_vm(vm=self.server2,
                                 vm_keypair=self.instance_keypair,
                                 ip_to_ping=self.server1_ip)
+
+    def try_to_update_router_to_centralized(self, router_id):
+        # Change admin_state_up to False
+        # and then try to set distributed to False
+        self.os_conn.neutron.update_router(router_id,
+                                           {'router': {
+                                            'admin_state_up': False}})
+
+        with pytest.raises(NeutronClientException) as e:
+            self.os_conn.neutron.update_router(router_id,
+                                           {'router': {
+                                            'distributed': False}})
+            err_msg = 'Distributed parameter was updated, but in case of'
+            err_msg = err_msg + ' dvr feature that should not be possible!'
+            assert False, err_msg
```

Comment:
`pytest.raises` context manager will fail if there is no raised exception inside it